### PR TITLE
Update new Mac Developer Certificate (ga)

### LINKED ISSUE #3912: [BUG] GA couldn't build macOS version
### Description
During working on the Apple Sign-In feature, we revoked some certificates, which causes the failure of Github Action.
```
error: No signing certificate "Mac Development" found: No "Mac Development" signing certificate matching team ID "B227VTMZ94" with a private key was found. (in target 'TogglDesktopLibrary' from project 'TogglDesktopLibrary')
```

We should cherry-pick the fixed commit on this PR and fix it on the master since Apple Sign In could be late to be merged into the master branch

### 💻 Environment
Platform: macOS

### 💯 Expected behavior
- GA on macOS runs without errors

### PR DESCRIPTION
**Pull request by NghiaTranUIT**
### 📒 Description
Use proper developer certificate for archiving the macOS build

### 🕶️ Types of changes
**Bug fix** (non-breaking change which fixes an issue)

### 🤯 List of changes
- Cherry-pick https://github.com/toggl-open-source/toggldesktop/pull/3879/commits/e27d34be079295b3261f93ff15936303a0a33e43 from Apple Sign-in PR

### 👫 Relationships
Closes #3912

### 🔎 Review hints
- Check whether or not the GA is success 👍

### CODE REVIEW

**Comment by skel35** [comment]
@NghiaTranUIT could we delete the unused certificate? to make sure there is no confusion in the future

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -46,13 +46,13 @@ jobs:
         security default-keychain -s ~/Library/Keychains/build.keychain
 
         # Mac Developer Certificate
-        base64 -D <<< "${{ secrets.CERT_MACOS_DEVELOPMENT }}" > Bundle_cert_macos_development.p12
-        security import ./Bundle_cert_macos_development.p12 -k ~/Library/Keychains/build.keychain -P ${{ secrets.CERT_MACOS_DEVELOPMENT_PASSWORD }} -T /usr/bin/codesign
+        base64 -D <<< "${{ secrets.CERT_MACOS_DEVELOPER }}" > Bundle_cert_macos_developer.p12
```
Comment (NghiaTranUIT) [inline .github/workflows/main.yml]: Replace expired cert with MAC OS Developer cert 👍

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -46,13 +46,13 @@ jobs:
...
+        security import ./Bundle_cert_macos_developer.p12 -k ~/Library/Keychains/build.keychain -P ${{ secrets.CERT_MACOS_DEVELOPER_PASSWORD }} -T /usr/bin/codesign
 
         # Application Certificate
         base64 -D <<< "${{ secrets.CERT_MACOS_APPLICATION }}" > Bundle_cert_macos_distribution.p12
```
Comment (NghiaTranUIT) [inline .github/workflows/main.yml]: Keep this certificate since it's for non-MAS distribution 👍

File: .github/workflows/main.yml (hunk 1 above)
Comment (skel35) [reply .github/workflows/main.yml]: 👍

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -46,13 +46,13 @@ jobs:
...
         security import ./Bundle_cert_macos_distribution.p12 -k ~/Library/Keychains/build.keychain -P ${{ secrets.CERT_MACOS_APPLICATION_PASSWORD }} -T /usr/bin/codesign
-        
+
         # Installer Certificate
```
Comment (NghiaTranUIT) [inline .github/workflows/main.yml]: This one is for MAS Submission, which we use later 👍

**Pull request merged**